### PR TITLE
Add newsletter month/year time frame option

### PR DESCRIPTION
**Pull request by jeffbargmann**
Howdy!

I'd like to send my newsletter monthly. I noticed there's a scheduling option for "monthly" in the scheduler -- "time frame" however can only do "30 days", which will miss a day some months. Added "monthly" time frame option to match what's availability in scheduler, and added "yearly" as well for good measure, for folks who may want to do yearly. Yeh neh?

Also: Moved "hourly" to the top to keep list chron; not sure what you think about the menu ordering?

Cheers 🙌

### CODE REVIEW

File: plexpy/newsletters.py [hunk 1]
```diff
@@ -386,6 +386,10 @@ def __init__(self, newsletter_id=None, newsletter_id_name=None, config=None, ema
         if self.start_date is None:
             if self.config['time_frame_units'] == 'days':
                 self.start_date = self.end_date.shift(days=-self.config['time_frame']+1).floor('day')
+            elif self.config['time_frame_units'] == 'months':
+                self.start_date = self.end_date.shift(months=-self.config['time_frame'], days=1).floor('day')
+            elif self.config['time_frame_units'] == 'years':
+                self.start_date = self.end_date.shift(years=-self.config['time_frame'], days=1).floor('day')
```
Comment (jeffbargmann) [inline plexpy/newsletters.py:392]: Included ", day=1" to crimp to specific year, a la 1/1/2020-12/31/2020, yes?

**Comment by jeffbargmann** [comment]
image

Worth noting: I'm operating on assumption the above doesn't miss a beat on days under 31 days. I'd considered adjusting phrase "31st" to "Last day" in this UI to be less ambiguous, but, (1) wasn't sure how to go about (2) this make the field longer. Anywho, thoughts?

In theory, perhaps this is all fubar, as on days < 31 days this would trigger "as soon as passed"... which would be the 1st of next month. So date readout would be for the 1st of next month. Perhaps not possible to send end of month newsletter 🤔 Lmk if I can help?

**Comment by JonnyWong16** [comment]
Pull requests must be made to the `nightly` branch.

**Comment by jeffbargmann** [comment]
👍 @JonnyWong16 closing this one. Thanks for the heads up

**Pull request closed without merging**